### PR TITLE
sonar-scanner (pr and master post merge)

### PR DESCRIPTION
**Pull request by Sybrand**
- run sonar-scanner on pr
- run sonar-scanner post merge

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -31,27 +31,17 @@ jobs:
...
-          export SONAR_SCANNER_VERSION=4.2.0.1873
-          export SONAR_SCANNER_HOME=$HOME/.sonar/sonar-scanner-$SONAR_SCANNER_VERSION-linux
-          curl --create-dirs -sSLo $HOME/.sonar/sonar-scanner.zip https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-$SONAR_SCANNER_VERSION-linux.zip
-          unzip -o $HOME/.sonar/sonar-scanner.zip -d $HOME/.sonar/
-          export PATH=$SONAR_SCANNER_HOME/bin:$PATH
-          export SONAR_SCANNER_OPTS="-server"
-          echo "---Run tests with coverage---"
-          ORIGINS=testorigin coverage run --source=. -m pytest
           coverage report
           coverage xml -o coverage-reports/coverage-report.xml
```
Comment (Kyubinhan) [inline .github/workflows/integration.yml:41]: what happens to this file `coverage-report.xml`? it gets uploaded to Sonarcloud or something?

File: .github/workflows/static_analysis.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install pipenv
+        uses: dschep/install-pipenv-action@v1
+
+      - name: Install Remaining Dependencies
+        run: pipenv install --dev --system
+
+      - name: Run unit tests with coverage
```
Comment (Kyubinhan) [inline .github/workflows/static_analysis.yml:33]: Can you tell me why we are running tests twice?

File: .github/workflows/integration.yml (hunk 1 above)
Comment (Sybrand) [reply .github/workflows/integration.yml:41]: Yep - sonar-runner pushes it to Sonarcloud.

File: .github/workflows/static_analysis.yml (hunk 2 above)
Comment (Sybrand) [reply .github/workflows/static_analysis.yml:33]: Sonarcloud doesn't make the report we've run for the PR apply to master on merge - so we have to re-run it against the master branch after merging.

**Pull request merged**